src/relay/ir/type_functor.cc

```diff
@@ -44,6 +44,29 @@ void TypeVisitor::VisitType_(const TypeRelationNode* op) {
   }
 }
 
+void TypeVisitor::VisitType_(const GlobalTypeVarNode* op) {
+}
+
+void TypeVisitor::VisitType_(const TypeCallNode* op) {
+  this->VisitType(op->func);
+  for (const Type& t : op->args) {
+    this->VisitType(t);
+  }
+}
+
+void TypeVisitor::VisitType_(const TypeDataNode* op) {
+  this->VisitType(op->header);
+  for (const auto& v : op->tv) {
+    this->VisitType(v);
+  }
+
```

Comment:
Not clear whether constructors should be visited, also not clear whether TypeData should really _be_ a type that gets visited, as opposed to a node that is only kept in a module (mapping from GlobalTypeVar -> TypeData). Would be curious as to others' thoughts on whether TypeData should be a Type.

Reply:
I wrote the code in a hurry, so should think about it.
I see no reason to visit it as well.

Reply:
I guess it's possible that a pass might want to do some transformation on ADT definitions? It could still access them through the module though (or is there a reason why it wouldn't be able to?)